lib/bitcoin/BitcoinClient.ts

```diff
@@ -89,14 +92,44 @@ export default class BitcoinClient {
     const transaction = await this.createTransaction(transactionData, feeInSatoshis);
     const rawTransaction = transaction.serialize();
 
-    const request = {
-      method: 'sendrawtransaction',
-      params: [
-        rawTransaction
-      ]
-    };
+    return this.broadcastTransactionRpc(rawTransaction);
+  }
 
-    return this.rpcCall(request, true);
+  /**
+   * Broadcasts the specified lock transaction.
+   *
+   * @param bitcoinLockTransaction The transaction object.
+   */
+  public async broadcastLockTransaction (bitcoinLockTransaction: BitcoinLockTransactionModel): Promise<string> {
+    if (!(bitcoinLockTransaction.transactionObject instanceof Transaction)) {
+      throw new Error('Transaction property is not of bitcore-lib.Transaction');
+    }
+
+    const transaction: Transaction = bitcoinLockTransaction.transactionObject;
+    const rawTransaction = transaction.serialize();
+
+    return this.broadcastTransactionRpc(rawTransaction);
+  }
+
+  /**
+   * Creates (and NOT broadcast) a lock transaction using the funds for the linked wallet.
+   *
+   * NOTE: if the linked wallet outputs are spent then this transaction cannot be broadcasted. So broadcast
+   * this transaction before spending from the wallet.
+   *
+   * @param lockAmountInSatoshis The amount to lock.
+   * @param lockUntilBlock  The block until the amount to lock to; the amount becomes spendable AT this block.
+   */
+  public async createLockTransaction (lockAmountInSatoshis: number, lockUntilBlock: number): Promise<BitcoinLockTransactionModel> {
+    const unspentCoins = await this.getUnspentOutputs(this.walletAddress);

```

Comment:
Consider merging this function with `createFreezeTransaction()` method.

Reply:
Will address it in the next iteration.